### DISCUSSION TITLE
iPad4 export - all message dates in html export set to Jan 01 2001, time 1:00:00 AM

### DISCUSSION THREAD

**Discussion by YossiBushida** [ReagentX/imessage-exporter#322] [Q&A] [answered]
Hi, I know this is related to old iPad4 (32bit device, old iOS), but maybe you know if this can be somehow easily fixed. All exported messages have same date and time - Jan 01 2001, time 1:00:00 AM. I guess the reason is older iOS versions have different structure of database kept in sms.db (different type of timestamp attribute?). Did anyone experience such problem and found a solution?

**Comment by ReagentX** [comment, marked as answer]
~~Older tables may use a different epoch, but this project only targets the current version of iOS/macOS.~~

 The timestamp issue was resolved in the [Chaparral Lotus](https://github.com/ReagentX/imessage-exporter/releases/tag/3.3.0) release.

**Reply by YossiBushida** [reply]
Hi, great thanks for quick answer, hint on epoch and very, very good tool :)

For those who care - a really ugly solution to export messages from older devices with proper dates seems to be:
1. first change sms.db manually - attributes 'date', 'date_read' and 'date_delivered' in the 'message' table should be multiplied by 1000000000 (using tool such as DB Browser for SQLite):
`UPDATE message
  SET
    date=date*1000000000,
    date_read=date_read*1000000000,
    date_delivered=date_delivered*1000000000`
2. Then just run imessage-exporter as usual
Take in mind that I checked this only on sms.db taken from 32bit iPad4, but still assume this should also be OK for other Apple old 32bit devices.
And of course apply changes to copy of sms.db downloaded from the device, do not make any manual changes on working copy of sms.db located on the device ;)

**Reply by ReagentX** [reply]
The timestamp issue was resolved in the [Chaparral Lotus](https://github.com/ReagentX/imessage-exporter/releases/tag/3.3.0) release.
